cluster/cluster.go

```diff
@@ -383,3 +394,21 @@ func ConfigureCluster(
 	}
 	return nil
 }
+
+func cleanControlNodes(ctx context.Context, kubeCluster, currentCluster *Cluster, cpToDelete []*hosts.Host) error {
+	for _, toDeleteHost := range cpToDelete {
+		kubeClient, err := k8s.NewClient(kubeCluster.LocalKubeConfigPath, kubeCluster.K8sWrapTransport)
```

Comment:
it feels like a repetition of https://github.com/rancher/rke/blob/5b3fec01657570642dd3ae61029b1fe270679af9/cluster/reconcile.go#L102. Can we re-use the same cleanup code logic?

Reply:
You can pass a custom message for the control node use case